doc: Drop description of LogError messages as fatal (#30361)
bitcoin/bitcoin: doc/developer-notes.md

  attacker to fill up storage. Note that `LogPrintf(fmt, params...)` is
  a deprecated alias for `LogInfo`.

> - `LogError(fmt, params...)` should be used in place of `LogInfo` for severe
>   errors the node admin will need to address (e.g., failure to write data).

Reviewer: nit: *`instead of`* seems [slightly more common](https://www.italki.com/en/post/question-107941) than *`in place of`* + `that` to streamline readability + `needs to address` to imply urgency:
```suggestion
- `LogError(fmt, params...)` should be used instead of `LogInfo` for severe errors
  that the node admin needs to address (e.g., failure to write data).
```

- `LogWarning(fmt, params...)` should be used in place of `LogInfo` for
  unexpected conditions indicating potentially severe problems the node admin
  should address (e.g. system time appears to be wrong, unknown soft fork
  appears to have activated).

> - `LogTrace(BCLog::CATEGORY, fmt, params...)` should be used in place of

Reviewer: nit: the logger order is weird, why debug -> info -> error -> warning -> trace?
Especially after announcing `LogInfo`, `LogDebug`, `LogTrace`, `LogWarning` and `LogError` in the intro.
